codility_training/lessons.lesson02.Arrays.CyclicRotation/khw970421.js

```diff
@@ -8,3 +9,19 @@ function solution(A, K) {
     }
     return b;
 }
+//방법2
+function solution1(A, K) {
+    const size = A.length;                          //배열의 크기 size
+    const shift = (A.length>K)?(K):K%A.length;      //움직인 횟수 shift K의값이 크기보다 클경우 반복되는 횟수 나머지만 계산
+
+    for(let i=0;i<shift;i++)                        //shift횟수만큼 반복
+    {
+        A.unshift(A[size-1]);                       //뒤의 값을 맨앞에 삽입
+        A.pop();                                    //맨뒤의 값을 삭제
+    }
+    return A;
```

Comment:
```suggestion
    return [ ...A.splice(size - K), ...A ]
```
unshift를 사용할 경우, 비용이 O(N ** 2) 까지 될 가능성이 있음
그래서 [splice](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Array/splice), [slice](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) 같은 것들을 사용하면 좋음